lib/rack/attack/store_proxy/active_support_redis_store_proxy.rb

```diff
@@ -13,16 +13,13 @@ def self.handle?(store)
         end
 
         def increment(name, amount = 1, options = {})
-          # #increment ignores options[:expires_in].
-          #
-          # So in order to workaround this we use #write (which sets expiration) to initialize
-          # the counter. After that we continue using the original #increment.
-          if options[:expires_in] && !read(name)
-            write(name, amount, options)
-
-            amount
-          else
-            super
+          rescuing do
+            with do |conn|
+              conn.multi do
+                conn.incrby(name, amount)
+                conn.expire(name, options[:expires_in]) if options[:expires_in]
+              end.first
+            end
```

Comment:
See description in https://github.com/kickstarter/rack-attack/pull/385 when this one done.

> [..] don't want to depend on any internals of externally developed cache store. We only wrap/proxy/adapt to the public interface using delegation.

Reply:
Similar comment here: can be optimized by not `read`ing for `redis-activesupport` >= `v5.2`, which added expiration support for `#increment` https://github.com/redis-store/redis-activesupport/commit/a794a315d658d8daae40370a23e6240a46471ee1.